pandas/tests/indexes/period/test_astype.py

```diff
@@ -164,3 +166,24 @@ def test_period_astype_to_timestamp(self):
         res = pi.astype("datetime64[ns, US/Eastern]", how="end")
         tm.assert_index_equal(res, exp)
         assert res.freq == exp.freq
+
+    def test_astype_tz_conversion_roundtrip(self):
+        # GH 35973
+        vals = {
+            "timezones": date_range(
+                "2020-08-30", freq="d", periods=2, tz="Europe/London"
+            )
+        }
+        df = DataFrame(vals)
+
+        # test UTC inferred object to specified tz
```

Comment:
you are converting a bunch of times but only asserting one.

also pls paramaterize this.

Reply:
we also have a tz_aware_fixture that you can test on

Reply:
done, although couldn't find the fixture.